src/joystick/SDL_gamecontroller.c

```diff
@@ -1798,6 +1814,26 @@ static SDL_bool SDL_GetControllerMappingFilePath(char *path, size_t size)
 #endif
 }
 
+static void SDLCALL
+SDL_JoystickHidapiVerticalJoyConsChanged(void *userdata, const char *name, const char *oldValue, const char *hint)
+{
+    SDL_HIDAPI_vertical_joycons = SDL_GetStringBoolean(hint, SDL_FALSE);
+
+    Uint16 vendor;
+    Uint16 product;
+    for (SDL_GameController *controller = SDL_gamecontrollers; controller; controller = controller->next) {
```

Comment:
Please lose this C99'ism (that for loop initial declaration)

Reply:
This would work, but isn't thread-safe and would override any custom controller mapping. For now let's remove this code and say this hint is only read at initialization and call it good.

Reply:
> This would work, but isn't thread-safe and would override any custom controller mapping. For now let's remove this code and say this hint is only read at initialization and call it good.

Updating of the mapping has been removed.